src/modules/secfilter/secfilter_hdr.c

```diff
@@ -0,0 +1,294 @@
+			{
+				LM_ERR("Possible SQLi detected in to user (%s)\n", val);
+				pkg_free(val);
+				return 0;
+			}
+
+			pkg_free(val);
+			val = (char*)pkg_malloc(parsed_uri.host.len+1);
+			if (val==NULL)
+			{
+				LM_ERR("Cannot allocate pkg memory\n");
+				return -1;
+			}
+			strncpy(val, parsed_uri.host.s, parsed_uri.host.len);
+			val[parsed_uri.host.len] = '\0';
+
+			if (strstr(val, "'") || strstr(val, "\"") || strstr(val, "--") || strstr(val, "=") || strstr(val, "#") || strstr(val, "+") || strstr(val, "%27") || strstr(val, "%24") || strstr(val, "%60"))
+			{
+				LM_ERR("Possible SQLi detected in to domain (%s)\n", val);
+				pkg_free(val);
+				return 0;
+			}
+	        }
+	}
+
+	if (val) pkg_free(val);
+	return 1;
+}
+
+/* Search for illegal characters in From header */
+int check_sqli_from(struct sip_msg *msg)
+{
+	struct to_body *from;
+	struct sip_uri parsed_uri;
+	char *val = NULL;
+
+	if (msg==NULL) return -1;
+	if(parse_from_header(msg) < 0) return -1;
+	if (msg->from==NULL || msg->from->body.s==NULL) return -1;
+
+	from = get_from(msg);
+	if (from != NULL)
+	{
+		val = (char*)pkg_malloc(from->display.len+1);
+		if (val==NULL)
+		{
+			LM_ERR("Cannot allocate pkg memory\n");
+			return -1;
+		}
+		strncpy(val, from->display.s, from->display.len);
+		val[from->display.len] = '\0';
+
+		if (strstr(val, "'") || (strstr(val, "\"") && count_chars(val, '"') != 2) || strstr(val, "--") || strstr(val, "=") || strstr(val, "#") || strstr(val, "%27") || strstr(val, "%24") || strstr(val, "%60"))
+		{
+			LM_ERR("Possible SQLi detected in from name (%s)\n", val);
+			pkg_free(val);
+			return 0;
+		}
+
+	        if (parse_uri(from->uri.s, from->uri.len, &parsed_uri)==0)
+	        {
+			pkg_free(val);
+			val = (char*)pkg_malloc(parsed_uri.user.len+1);
+			if (val==NULL)
+			{
+				LM_ERR("Cannot allocate pkg memory\n");
+				return -1;
+			}
+			strncpy(val, parsed_uri.user.s, parsed_uri.user.len);
+			val[parsed_uri.user.len] = '\0';
+
+			if (strstr(val, "'") || strstr(val, "\"") || strstr(val, "--") || strstr(val, "=") || strstr(val, "#") || strstr(val, "%27") || strstr(val, "%24") || strstr(val, "%60"))
+			{
+				LM_ERR("Possible SQLi detected in from user (%s)\n", val);
+				pkg_free(val);
+				return 0;
+			}
+
+			pkg_free(val);
+			val = (char*)pkg_malloc(parsed_uri.host.len+1);
+			if (val==NULL)
+			{
+				LM_ERR("Cannot allocate pkg memory\n");
+				return -1;
+			}
+			strncpy(val, parsed_uri.host.s, parsed_uri.host.len);
+			val[parsed_uri.host.len] = '\0';
+
+			if (strstr(val, "'") || strstr(val, "\"") || strstr(val, "--") || strstr(val, "=") || strstr(val, "#") || strstr(val, "+") || strstr(val, "%27") || strstr(val, "%24") || strstr(val, "%60"))
+			{
+				LM_ERR("Possible SQLi detected in from domain (%s)\n", val);
+				pkg_free(val);
+				return 0;
+			}
+	        }
+	}
+
+	if (val) pkg_free(val);
+	return 1;
+}
+
+
+/* Search for illegal characters in Contact header */
+int check_sqli_contact(struct sip_msg *msg)
```

Comment:
See above, could be done with PVs and transformations: $ct{uri.domain}, $ct{uri.user}